Add new events and improve telemetry code (#1584)
adap/flower: doc/source/telemetry.md

Flower telemetry collects the following metrics:

**Flower version.** Understand which versions of Flower are currently being used. This helps us to decide whether we should invest effort into releasing a patch version for an older version of Flower or instead use the bandwidth to build new features.

**Operating system.** Enables us to answer questions such as: *Should we create more guides for Linux, macOS, or Windows?*

> **Python version.** Knowing the Python version helps us, for example, decide whether we should invest effort into supporting old versions of Python or stop supporting them and start taking advantage of new Python features.

Reviewer: ```suggestion
**Python version.** Knowing the Python version helps us, for example, to decide whether we should invest effort into supporting old versions of Python or stop supporting them and start taking advantage of new Python features.
```

> **CPU/GPU properties.** Knowing some general data about CPU/GPU properties helps us to understand the hardware environment Flower is being used in and whether we should, for example, put more effort into supporting low-resource environments.

Reviewer: ```suggestion
**Hardware properties.** Understanding the hardware environment that Flower is being used in helps to decide whether we should, for example, put more effort into supporting low-resource environments.
```

> **Execution mode.** Knowing in which execution mode Flower starts enables us to understand how much certain features are used and better prioritize based on that.

Reviewer: ```suggestion
**Execution mode.** Knowing what execution mode Flower starts in enables us to understand how heavily certain features are being used and better prioritize based on that.
```

> We will not collect any personally identifiable information. If you think any of the metrics collected could be misused in any way, please [get in touch with us](#how-to-contact-us). We will update this page to reflect any changes to the metrics collected and publish all changes in the changelog.

Reviewer: ```suggestion
We will not collect any personally identifiable information. If you think any of the metrics collected could be misused in any way, please [get in touch with us](#how-to-contact-us). We will update this page to reflect any changes to the metrics collected and publish changes in the changelog.
```
